source/common/network/utility.cc

```diff
@@ -246,7 +246,8 @@ bool Utility::isLocalConnection(const Network::ConnectionSocket& socket) {
   // connection. However, this is a rare exception and such assumption results in big
   // performance optimization.
   if (remote_address->type() == Envoy::Network::Address::Type::Pipe ||
-      remote_address == socket.localAddress() || isLoopbackAddress(*remote_address)) {
+      remote_address->equalsExceptPort(*socket.localAddress()) ||
+      isLoopbackAddress(*remote_address)) {
```

Comment:
This is the meat of the change

Reply:
Instead of trying to do this via an interface method, I would just do it here directly via testing if the addresses are IP addresses, and if so, then testing if the IP addresses are equal. Also, please update the comment above to talk about what is considered a local connection (similar to your PR description).

Reply:
done